src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -733,7 +733,8 @@ private void EmitArgument(BoundExpression argument, RefKind refKind)
                     if (unexpectedTemp != null)
                     {
                         // interestingly enough "ref dynamic" sometimes is passed via a clone
-                        Debug.Assert(argument.Type.IsDynamic(), "passing args byref should not clone them into temps");
+                        // receiver of a ref field can be cloned too
+                        Debug.Assert(argument.Type.IsDynamic() || argument is BoundFieldAccess { FieldSymbol.RefKind: RefKind.Ref }, "passing args byref should not clone them into temps");
```

Comment:
Is it possible to get here with `FieldSymbol.RefKind: RefKind.RefReadonly` as well?

Reply:
> Is it possible to get here with `FieldSymbol.RefKind: RefKind.RefReadonly` as well?

I think we will end up in the `case` above. I added a test for the scenario and it confirmed the theory. However, I am going to relax the assert anyway, just in case.